### PR TITLE
feat(iron-bank): added iron bank to Zapper 

### PR DESCRIPTION
**Pull request by reganwrench**
## Description

Adding integration of Iron Bank (ETH, FTM, AVAX)

## Checklist

- [v] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)

## How to test?

1. Checked TVL against Iron Bank's [website](https://app.ib.xyz/markets/Avalanche)
2. Check account balance with a sample from [USDC market on Ethereum](https://etherscan.io/address/0x76Eb2FE28b36B3ee97F3Adae0C69606eeDB2A37c)
3. Check account balance with a sample from [WFTM market on Fantom](https://ftmscan.com/address/0xd528697008ac67a21818751a5e3c58c8dae54696)
4. Check account balance with a sample from [USDT.e market on Avalanche](https://snowtrace.io/address/0xceb1ce674f38398432d20bc8f90345e91ef46fd3)

### CODE REVIEW

**Comment by immasandwich** [comment]
Hello @reganwrench !

Instead of copying the helpers from `src/apps/compound/helpers`, please reference those directly in your fetcher classes (the helpers are intended to be used for Compound and any fork of Compound contracts). You'll also need `CompoundAppModule` to the `imports` array in your `IronBankModule`.

**Comment by reganwrench** [comment]
> Hello @reganwrench !
> 
> Instead of copying the helpers from `src/apps/compound/helpers`, please reference those directly in your fetcher classes (the helpers are intended to be used for Compound and any fork of Compound contracts). You'll also need `CompoundAppModule` to the `imports` array in your `IronBankModule`.

ser, i have pushed an update that uses the helpers from compound directly. thank you.

File: src/apps/iron-bank/avalanche/ironBank.balance-fetcher.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const [supplyBalances, borrowBalances] = await Promise.all([
+      this.getSupplyBalances(address),
+      this.getBorrowBalances(address)
+    ]);
+
+    const meta = this.compoundLendingMetaHelper.getMeta({ balances: [...supplyBalances, ...borrowBalances] });
+    const lendingProduct = { label: 'Lending', assets: [...supplyBalances, ...borrowBalances], meta };
+
+    return presentBalanceFetcherResponse([lendingProduct]);
+}
```
Comment (immasandwich) [inline src/apps/iron-bank/avalanche/ironBank.balance-fetcher.ts:62]: I think you have some lint errors, run `pnpm lint:fix` to resolve.

File: src/apps/iron-bank/avalanche/ironBank.supply.token-fetcher.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @Inject(IronBankContractFactory) private readonly ironBankContractFactory: IronBankContractFactory,
+    @Inject(CompoundSupplyTokenHelper) private readonly compoundSupplyTokenHelper: CompoundSupplyTokenHelper,
+  ) {}
+
+  async getPositions() {
+    return this.compoundSupplyTokenHelper.getTokens({
+      network,
+      appId,
+      groupId,
+      comptrollerAddress: '0x2eE80614Ccbc5e28654324a66A396458Fa5cD7Cc',
```
Comment (immasandwich) [inline src/apps/iron-bank/avalanche/ironBank.supply.token-fetcher.ts:28]: Lowercase address 🙏

File: src/apps/iron-bank/fantom/ironBank.supply.token-fetcher.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @Inject(IronBankContractFactory) private readonly ironBankContractFactory: IronBankContractFactory,
+    @Inject(CompoundSupplyTokenHelper) private readonly compoundSupplyTokenHelper: CompoundSupplyTokenHelper,
+  ) {}
+
+  async getPositions() {
+    return this.compoundSupplyTokenHelper.getTokens({
+      network,
+      appId,
+      groupId,
+      comptrollerAddress: '0x4250A6D3BD57455d7C6821eECb6206F507576cD2',
```
Comment (immasandwich) [inline src/apps/iron-bank/fantom/ironBank.supply.token-fetcher.ts:28]: Lowercase 🙏

**Comment by immasandwich** [comment]
@reganwrench Last few requests:

- Ensure filenames use `iron-bank` instead of `ironBank` (this includes the JSON ABI files in `src/apps/iron-bank/contracts/abi`, and then re-generate your contract factory).
- Fix the lint errors
- Lowercase comptroller addresses

Then we're ready! I'll slap an approval on this too right now.

**Review by immasandwich — APPROVED** [review]

File: src/apps/iron-bank/avalanche/ironBank.balance-fetcher.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import { Inject, forwardRef } from '@nestjs/common';
```
[the remark this answers is not in the document]
Comment (reganwrench) [reply src/apps/iron-bank/avalanche/ironBank.balance-fetcher.ts]: fixed

File: src/apps/iron-bank/avalanche/ironBank.supply.token-fetcher.ts (hunk 2 above)
Comment (reganwrench) [reply src/apps/iron-bank/avalanche/ironBank.supply.token-fetcher.ts]: done

File: src/apps/iron-bank/ethereum/ironBank.supply.token-fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    @Inject(IronBankContractFactory) private readonly ironBankContractFactory: IronBankContractFactory,
+    @Inject(CompoundSupplyTokenHelper) private readonly compoundSupplyTokenHelper: CompoundSupplyTokenHelper,
+  ) {}
+
+  async getPositions() {
+    return this.compoundSupplyTokenHelper.getTokens({
+      network,
+      appId,
+      groupId,
+      comptrollerAddress: '0xAB1c342C7bf5Ec5F02ADEA1c2270670bCa144CbB',
```
[the remark this answers is not in the document]
Comment (reganwrench) [reply src/apps/iron-bank/ethereum/ironBank.supply.token-fetcher.ts]: done

File: src/apps/iron-bank/fantom/ironBank.supply.token-fetcher.ts (hunk 3 above)
Comment (reganwrench) [reply src/apps/iron-bank/fantom/ironBank.supply.token-fetcher.ts]: done

**Comment by reganwrench** [comment]
> @reganwrench Last few requests:
> 
> * Ensure filenames use `iron-bank` instead of `ironBank` (this includes the JSON ABI files in `src/apps/iron-bank/contracts/abi`, and then re-generate your contract factory).
> * Fix the lint errors
> * Lowercase comptroller addresses
> 
> Then we're ready! I'll slap an approval on this too right now.

all done. thank you ser

**Comment by SamIam-0x** [comment]
Hello @reganwrench ! 

Looks like you did not provide an address when submitting your PR. I recognize it may have seemed kind of random that we were asking for a wallet address with a PR submission, but we are actually looking for ways to reward Studio contributors in the future. 

Is there an address you’d want to provide? Could even be an empty address, if you do not have an doxxed address

Let me know! Also, if you’d like to privately provide it, feel free to shoot me a TG https://t.me/SamIAm_0x or DM on the Zapper Discord.

Thanks!
Sam

**Comment by reganwrench** [comment]
@SamIam-0x 
thank you ser
0x51e5Ba55bF3D66A78108DA9D0444178880139881

**Pull request merged**